### PR TITLE
Implement --version in click

### LINKED ISSUE #6757: [CT-1926] Add support for --version to click cli
Using [click's options and callbacks](https://click.palletsprojects.com/en/8.1.x/options/#callbacks-and-eager-options), add support for `--version` to the new click cli.

NOTE: We use dbt --version as part of the Code Quality check in GHA.  Since it's not working currently that has been disabled [here](https://github.com/dbt-labs/dbt-core/commit/1ead5ac2b29220d6a06d2425f1aabeeead51db60).  Please un-comment that part of the workflow when this ticket is ready to go.

### PR DESCRIPTION
**Pull request by stu-k**
resolves #6757

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [X] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [X] I have run this code in development and it appears to resolve the stated issue
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [X] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [X] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: core/dbt/cli/params.py
Comment (stu-k) [inline core/dbt/cli/params.py:378]: I'm not sure if this is the best place for this function, but I wanted to use click's callbacks for this since we're already so invested in click.

File: tests/unit/test_dbt_runner.py
Comment (stu-k) [inline tests/unit/test_dbt_runner.py:30]: Is this the best way to handle this exception? We do want to exit early when using version so no further commands are run, but putting this here felt a bit strange.

**Review by iknox-fa — APPROVED** [review]
LGTM

**Comment by iknox-fa** [comment]
FYI -- [This PR](https://github.com/dbt-labs/dbt-core/pull/6785) will have to be merged before the tests will pass because we actually use the version command in our CI pipeline.

File: tests/unit/test_dbt_runner.py
Comment (MichelleArk) [reply tests/unit/test_dbt_runner.py:30]: Given that `dbtRunner` is meant for programmatic invocations - I think we should handle this in `dbtRunner` itself (as opposed to the tests) so that callers don't need to catch `click` exceptions. Perhaps in a new `except` block here: https://github.com/dbt-labs/dbt-core/compare/feature/click-cli#diff-43832707bfec8cdaa49c13f004d9b3296585827f7e68ef07fba7ad1d5e6eb7c6R55-R58

File: tests/unit/test_dbt_runner.py
Comment (stu-k) [reply tests/unit/test_dbt_runner.py]: Handled the exit exception in `dbtRunner` as per @MichelleArk's suggestion.

**Review by ChenyuLInx — COMMENTED** [review]
Changes looks good, although it seems also contain a few other changes that haven't been merged yet? maybe we should wait for those to merge before merging this?

**Comment by stu-k** [comment]
@ChenyuLInx yeah I messed something up when trying to merge `feature/click-cli` into this branch. Give me a moment to fix it 🤦

**Comment by stu-k** [comment]
Closing in favor of #6802

**Pull request closed without merging**
